### PR TITLE
Bug 1266229 - Pulse ingestion fixes and updates

### PR DESCRIPTION
**Pull request by camd**
These changes were discovered to be needed after direct testing
against a Task Cluster exchange from the ``taskcluster-treeherder`` project with Pulse.

-Made some schema changes to be more precise for several fields
-Modified to job_loader to be more resilient to optional data being
missing
-Change the ``name`` field for job types and job groups to allow 100
  characters because some names were being truncated

---
This change is [Reviewable](https://reviewable.io/reviews/mozilla/treeherder/1433)

### CODE REVIEW

File: treeherder/etl/job_loader.py [hunk 1]
```diff
@@ -116,14 +135,13 @@ def _get_artifacts(self, job, job_guid):
...
         if "jobInfo" in job:
-
             ji = job["jobInfo"]
             job_details = []
             if "summary" in ji:
                 job_details.append({
                     "content_type": "raw_html",
-                    "value": ji["summary"]["text"],
-                    "title": ji["summary"]["label"]
+                    "value": ji["summary"],
```
Comment (camd) [inline treeherder/etl/job_loader.py]: ends up this wasn't matching the schema.  oops...

File: schemas/pulse-job.yml [hunk 2]
```diff
@@ -61,12 +61,12 @@ properties:
               This could be the organization or the individual git username
               depending on who owns the repo.
             type: "string"
-            pattern: "^[0-9a-f]+$"
+            pattern: "^[A-Za-z0-9_-]+$"
```
Comment (wlach) [inline schemas/pulse-job.yml]: Can you not just \w here (or even everywhere?). https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions#special-word

File: treeherder/etl/job_loader.py [hunk 3]
```diff
@@ -231,9 +249,9 @@ def _get_platform(self, platform_src):
     def _get_machine(self, job):
         machine = "unknown"
         if "buildMachine" in job:
-            machine = job["buildMachine"]["name"]
+            machine = job["buildMachine"].get("name", machine)
         if "runMachine" in job:
```
Comment (wlach) [inline treeherder/etl/job_loader.py]: Should this be `if` or `elif`?

File: treeherder/model/models.py [hunk 4]
```diff
@@ -308,7 +308,7 @@ def truncate(self, skip_list=None):
 class JobGroup(models.Model):
     id = models.AutoField(primary_key=True)
     symbol = models.CharField(max_length=10, default='?', db_index=True)
-    name = models.CharField(max_length=50, db_index=True)
+    name = models.CharField(max_length=100, db_index=True)
```
Comment (wlach) [inline treeherder/model/models.py]: Could you put these schema changes into a different commit (keep the same bug)?

File: schemas/pulse-job.yml (hunk 2 above)
Comment (camd) [inline schemas/pulse-job.yml]: ahh, good catch.  I can at least USE it.  I have to add in a couple extra characters here and there.  And a couple cases there underscore are not acceptable (``job_guid``) but other than that, this helps keep the patterns shorter.  thanks! :)

File: treeherder/etl/job_loader.py (hunk 3 above)
Comment (camd) [inline treeherder/etl/job_loader.py]: erp..  good catch.  changed to ``elif``

File: treeherder/model/models.py (hunk 4 above)
Comment (camd) [inline treeherder/model/models.py]: will do, sure.
